sdk/azidentity/azure_cli_credential_client.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+const (
+	DefaultSuffix        = "/.default"
+	AzureCLINotInstalled = "Azure CLI not installed"
+	AzNotLogIn           = "ERROR: Please run 'az login'"
+	WinAzureCLIError     = "'az' is not recognized"
+	invalidResourceError = "Resource is not in expected format. Only alphanumeric characters, '.', ':', '-', and '/' are allowed"
+)
+
+type azureCliAccessToken struct {
+	Token        string `json:"accessToken"`
+	ExpiresOn    string `json:"expiresOn"`
+	Subscription string `json:"subscription"`
+	Tenant       string `json:"tenant"`
+	TokenType    string `json:"tokenType"`
+}
+
+type ShellClient interface {
+	getAzureCliAccessToken(command string) ([]byte, string, error)
+}
+
+// AzureCliCredentialClient provides the base for authenticating with Azure Cli Credential.
+type AzureCliCredentialClient struct {
```

Comment:
I'm wondering whether this should have a pipeline to handle the token request and whether this should simple exist in aadIdentityClient. @jhendrixMSFT what do you think?

UPDATE: looking at the code I don't think this particular credential needs a pipeline since it simply executes a shell command